Emby.Server.Implementations/LiveTv/Listings/XmlTvListingsProvider.cs

```diff
@@ -67,110 +64,66 @@ private async Task<string> GetXml(ListingsProviderInfo info, CancellationToken c
+            if (info.Path.StartsWith("http", StringComparison.OrdinalIgnoreCase))
+            {
+                _logger.LogInformation("Downloading xmltv listings from {Path}", info.Path);
 
-            Directory.CreateDirectory(Path.GetDirectoryName(cacheFile));
+                using var response = await _httpClientFactory.CreateClient(NamedClient.Default).GetAsync(info.Path, cancellationToken).ConfigureAwait(false);
+                await using var stream = await response.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false);
 
-            using var response = await _httpClientFactory.CreateClient(NamedClient.Default).GetAsync(info.Path, cancellationToken).ConfigureAwait(false);
-            await using var stream = await response.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false);
-            await using (var fileStream = new FileStream(cacheFile, FileMode.CreateNew, FileAccess.Write, FileShare.None, IODefaults.CopyToBufferSize, FileOptions.Asynchronous))
-            {
-                await stream.CopyToAsync(fileStream, cancellationToken).ConfigureAwait(false);
+                return await UnzipIfNeededAndCopy(info.Path, stream, cacheFile, cancellationToken).ConfigureAwait(false);
             }
+            else
+            {
+                await using var stream = new FileStream(info.Path, FileMode.Open, FileAccess.Read, FileShare.Read, IODefaults.FileStreamBufferSize, FileOptions.Asynchronous);
 
-            return UnzipIfNeeded(info.Path, cacheFile);
+                return await UnzipIfNeededAndCopy(info.Path, stream, cacheFile, cancellationToken).ConfigureAwait(false);
+            }
         }
 
-        private string UnzipIfNeeded(ReadOnlySpan<char> originalUrl, string file)
+        private async Task<string> UnzipIfNeededAndCopy(string originalUrl, Stream stream, string file, CancellationToken cancellationToken)
         {
-            ReadOnlySpan<char> ext = Path.GetExtension(originalUrl.LeftPart('?'));
+            int index = originalUrl.IndexOf('?', StringComparison.CurrentCulture);
+            string ext = Path.GetExtension(index > -1 ? originalUrl.Remove(index) : originalUrl);
 
             if (ext.Equals(".gz", StringComparison.OrdinalIgnoreCase))
             {
                 try
                 {
-                    string tempFolder = ExtractGz(file);
-                    return FindXmlFile(tempFolder);
-                }
-                catch (Exception ex)
-                {
-                    _logger.LogError(ex, "Error extracting from gz file {File}", file);
-                }
-
-                try
-                {
-                    string tempFolder = ExtractFirstFileFromGz(file);
-                    return FindXmlFile(tempFolder);
+                    using var reader = new GZipStream(stream, CompressionMode.Decompress);
+                    await using var writer = File.Create(file);
```

Comment:
Why create this FileStream differently then the one on line 122?

Reply:
Good catch, the should be equal